aws/resource_aws_codebuild_project.go

```diff
@@ -380,15 +431,19 @@ func resourceAwsCodeBuildProjectRead(d *schema.ResourceData, meta interface{}) e
 
 	project := resp.Projects[0]
 
-	if err := d.Set("artifacts", flattenAwsCodebuildProjectArtifacts(project.Artifacts)); err != nil {
+	if err := d.Set("artifacts", flattenAwsCodeBuildProjectArtifacts(project.Artifacts)); err != nil {
 		return err
 	}
 
-	if err := d.Set("environment", schema.NewSet(resourceAwsCodeBuildProjectEnvironmentHash, flattenAwsCodebuildProjectEnvironment(project.Environment))); err != nil {
+	if err := d.Set("environment", schema.NewSet(resourceAwsCodeBuildProjectEnvironmentHash, flattenAwsCodeBuildProjectEnvironment(project.Environment))); err != nil {
 		return err
 	}
 
-	if err := d.Set("source", flattenAwsCodebuildProjectSource(project.Source)); err != nil {
+	if err := d.Set("source", flattenAwsCodeBuildProjectSource(project.Source)); err != nil {
+		return err
+	}
+
+	if err := d.Set("vpc_config", flattenAwsCodeBuildVpcConfig(project.VpcConfig)); err != nil {
```

Comment:
No matter how I sort subnets (and I've tried a few). I seem to get failing scenarios, but this line seems to be the indicator of the failure:
```
vpc_config is set to [map[security_group_ids:[sg-19a81766] vpc_id:vpc-61a22318 subnets:[subnet-d503a08f subnet-953c9fcf]]]
vpc_config is set to [map[vpc_id:vpc-61a22318 subnets:[subnet-d503a08f subnet-953c9fcf] security_group_ids:[sg-19a81766]]]
```

The object has the same values. The subnets are in the same order, but I get this error:

```
--- FAIL: TestAccAWSCodeBuildProject_basic (43.42s)
        testing.go:513: Step 0 error: After applying this step, the plan was not empty:

                DIFF:

                UPDATE: aws_codebuild_project.foo
                  vpc_config.0.subnets.0: "subnet-d503a08f" => "subnet-953c9fcf"
                  vpc_config.0.subnets.1: "subnet-953c9fcf" => "subnet-d503a08f"

                STATE:
```

Is it easier just go back to Set and calculate the hash for the difference?

Reply:
The `subnets` and `security_group_ids` can be `schema.TypeSet`, but you don't need a special hash function for them. Terraform core will "do the right thing" when comparing simple string sets.